### PR TITLE
Fixing issue when send synonyms as empty array

### LINKED ISSUE #204: Unable to `updateSettings` for an index
**Description**
I want to update and index settings using the script below

```php
...
$settings = $index->getSettings();
$index->updateSettings($settings);
```

**Expected behavior**
The settings would be successfully updated

**Current behavior**
An exception is thrown

```php
MeiliSearch\Exceptions\ApiException with message 'Json deserialize error: invalid type: sequence, expected a map at line 1 column 190'
```

If I use `updateSearchableAttributes` or `updateFilterableAttributes` it works as expected

**Environment (please complete the following information):**
- OS: [e.g. Mac OSX]
- MeiliSearch version: [e.g. v.0.21.1]
- meilisearch-php version: [e.g v0.19.0]

### PR DESCRIPTION
**Pull request by alallema**
This PR is a patch for this [issue](#204).
But it could be interesting to implement a complete JSONSerializer for this SDK see #209.
closed #204.

### CODE REVIEW

**Review by curquiza — COMMENTED** [review]
Could you add `Should be removed when implementing https://github.com/meilisearch/meilisearch-php/issues/209` in comment? :)

File: tests/Settings/SettingsTest.php [hunk 1]
```diff
@@ -157,4 +157,19 @@ public function testResetSettings(): void
         $this->assertIsArray($settings['sortableAttributes']);
         $this->assertEmpty($settings['sortableAttributes']);
     }
+
+    // Here to test for regressions related to https://github.com/meilisearch/meilisearch-php/issues/204.
```
Comment (curquiza) [inline tests/Settings/SettingsTest.php:161]: What do you mean "for regressions"?

Comment (alallema) [reply tests/Settings/SettingsTest.php:161]: That these is the new tests to ensure that this error will not happen again

**Review by curquiza — APPROVED** [review]
Well done! 🚀

**Comment by alallema** [comment]
bors merge

**Pull request merged**